zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -92,7 +93,16 @@ class EcommerceSpider(Args[EcommerceSpiderParams], BaseSpider):
     @classmethod
     def from_crawler(cls, crawler: Crawler, *args, **kwargs) -> scrapy.Spider:
         spider = super(EcommerceSpider, cls).from_crawler(crawler, *args, **kwargs)
-        spider.allowed_domains = [get_domain(spider.args.url)]
+        seed_url = spider.args.seed_url
+        if seed_url:
+            response = requests.get(seed_url)
+            urls = [url.strip() for url in response.text.split("\n")]
```

Comment:
Perhaps we should also validate URLs here using the _URL_PATTERN and maybe drop any malformed entries in the process.

Though maybe a better approach would be to error out and don't start the crawl since depending on the use case, it may be a waste to begin a crawl if even a single malformed URL was present (_due to human error)_.